src/main/java/it/pagopa/pn/paperchannel/middleware/queue/consumer/handler/RECAG008CMessageHandler.java

```diff
@@ -30,7 +29,8 @@ public class RECAG008CMessageHandler extends SendToDeliveryPushHandler {
     public Mono<Void> handleMessage(PnDeliveryRequest entity, PaperProgressStatusEventDto paperRequest) {
         return eventMetaDAO.findAllByRequestId(buildMetaRequestId(paperRequest.getRequestId()))
                 .collectList()
-                .filter(pnEventMetas -> this.correctPreviousEventMeta(pnEventMetas, paperRequest.getRequestId()))
+                .filter(pnEventMetas ->
+                        this.correctPreviousEventMeta(pnEventMetas, paperRequest.getRequestId(), entity, paperRequest))
                 .doOnNext(pnEventMetas -> log.info("[{}] Found correct previous states", paperRequest.getRequestId()))
```

Comment:
@Vallasc qui per evitare di inserire parametri nel metodo `correctPreviousEventMeta`, potresti aggiornare la condizione if (!ok) ritornando semplicemente il risultato della valutazione elRECAG012.isPresent() && elPNAG012.isPresent().

In caso in cui il ritorno è false, per tirare l'eccezione ti basterebbe aggiungere, sopra il `doOnNext` a riga 34, un `doOnDiscard` e qui lanci l'eccezione che ti serve avendo già tutti gli oggetti pronti